xicam/gui/models/treemodel.py

```diff
@@ -1,7 +1,8 @@
 from typing import Any
 from qtpy.QtCore import Qt, QAbstractItemModel, QModelIndex
 
-
+# TODO: refactor check_state (don't need custom methods, use setData)
+# TODO: refactor flags (don't need custom methods? use setData)
```

Comment:
flags are probably fine; lets fix check_state to use roles

Reply:
this is fixed